pkg/skaffold/initializer/deploy/kustomize.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deploy
+
+import (
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+)
+
+// kustomize implements deploymentInitializer for the kustomize deployer.
+type kustomize struct {
+	kustomizations []string
+	images         []string
+}
+
+// newKustomizeInitializer returns a kustomize config generator.
+func newKustomizeInitializer(kustomizations []string, potentialConfigs []string) *kustomize {
+	var images []string
+	for _, file := range potentialConfigs {
+		imgs, err := kubernetes.ParseImagesFromKubernetesYaml(file)
+		if err == nil {
+			images = append(images, imgs...)
+		}
+	}
+	return &kustomize{
+		images:         images,
+		kustomizations: kustomizations,
+	}
+}
+
+// deployConfig implements the Initializer interface and generates
+// a kustomize deployment config.
+func (k *kustomize) DeployConfig() latest.DeployConfig {
+	var kustomizeConfig *latest.KustomizeDeploy
+	// if we only have the default path, leave the config empty - it's cleaner
+	if len(k.kustomizations) == 1 && k.kustomizations[0] == deploy.DefaultKustomizePath {
+		kustomizeConfig = &latest.KustomizeDeploy{}
+	} else {
+		kustomizeConfig = &latest.KustomizeDeploy{
+			KustomizePaths: k.kustomizations,
```

Comment:
so, I'm a bit hesitant around this - 
typical users might have this setup:

```
├── base
│   ├── deployment.yaml
│   └── kustomization.yaml
├── dev
│   ├── kustomization.yaml
│   └── patch.yaml
├── prod
│   ├── kustomization.yaml
│   └── patch.yaml
├── skaffold-kustomize-args.yaml
└── skaffold.yaml
```

And then we get a skaffold.yaml like this: 

```
apiVersion: skaffold/v2beta1
kind: Config
metadata:
  name: kustomize
deploy:
  kustomize:
    paths:
    - base
    - dev
    - prod
```

So...I would consider doing this instead: 
1. we should skip `base` 
2. for each kustomize file we would generate a profile...

Could be done in a follow-up PR. But again, design doc, user journeys would be great to tackle.

Reply:
yeah, interesting. I wonder if we would want to prompt the user for their "default" kustomization so they don't always have to run with a profile? I'd imagine if users have a "dev" and a "prod" configuration, they'd clearly want "dev" as the default. this is good feedback though, I'm gonna follow this PR up with some improvements.